exact_string_matching/backward.py

```diff
@@ -116,3 +116,23 @@ def turbo_boyer_moore(t, w, n, m):
     else:
       i = i + shift
       memory = min(m - shift, match)
+
+def Q(i, k, m, w):
+  return (k <= i and w[(i-k+1):i] == w[(m-k+1):m]) or (k > i and w[1:i] != w[(m-i+1):m])
```

Comment:
Please put it as an inner function of boyer_moore_apostolico_giancarlo - this way you can remove m and w from the parameters list.

Moreover, is write w[something:m] supposed to omit the last element or should it be rather w[something:]?

Reply:
I belive it should omit first and last one

Reply:
I will add description of corse, just want to start with algorithm to check if it works

Reply:
Thank you very much, everything is working now.